scripts/forms-scaffolder.js

```diff
@@ -0,0 +1,430 @@
+    'Email',
+    'File Input',
+    'Image',
+    'Number Input',
+    'Panel',
+    'Radio Group',
+    'Reset Button',
+    'Submit Button',
+    'Telephone Input',
+    'Text',
+    'Text Input',
+  ];
+
+  return baseComponents.map((name) => ({
+    name,
+    value: name.toLowerCase().replace(/\s+/g, '-'),
+    filename: `_${name.toLowerCase().replace(/\s+/g, '-')}.json`,
+  }));
+}
+
+// Read customComponents array from mappings.js
+function readCustomComponents() {
+  const mappingsPath = path.join(__dirname, '../blocks/form/mappings.js');
+
+  try {
+    const mappingsContent = readFileSync(mappingsPath, 'utf-8');
+    const customComponentsRegex = /let customComponents = \[([^\]]*)\];/;
+    const match = mappingsContent.match(customComponentsRegex);
+
+    if (match) {
+      return match[1]
+        .split(',')
+        .map((comp) => comp.trim().replace(/['"]/g, ''))
+        .filter((comp) => comp.length > 0);
+    }
+    return [];
+  } catch (error) {
+    return [];
+  }
+}
+
+// Component name validation
+function validateComponentName(name) {
+  if (!name || typeof name !== 'string') {
+    return 'Component name is required';
+  }
+
+  if (name !== name.toLowerCase()) {
+    return 'Component name must be lowercase';
+  }
+
+  if (!/^[a-z][a-z0-9-]*$/.test(name)) {
+    return 'Component name must start with a letter and can only contain lowercase letters, numbers, and hyphens';
+  }
+
+  if (name.startsWith('-') || name.endsWith('-')) {
+    return 'Component name cannot start or end with a hyphen';
+  }
+
+  if (name.includes('__') || name.includes('_')) {
+    return 'Component name cannot contain underscores';
+  }
+
+  // Check if component already exists
+  const existingComponents = readCustomComponents();
+  if (existingComponents.includes(name)) {
+    return `Component '${name}' already exists. Please choose a different name.`;
+  }
+
+  return true;
+}
+
+// Create component files
+function createComponentFiles(componentName, baseComponent, targetDir) {
+  const files = {
+    js: `${componentName}.js`,
+    css: `${componentName}.css`,
+    json: `_${componentName}.json`,
+  };
+
+  // Create JS file
+  const jsContent = `/**
+ * Custom ${componentName} component
+ * Based on: ${baseComponent.name}
+ */
+export default async function decorate(fieldDiv, fieldJson) {
+  console.log('${emojis.gear} Decorating ${componentName} component:', fieldDiv, fieldJson);
+  
+  // TODO: Implement your custom component logic here
+  // You can access the field properties via fieldJson.properties
+  
+  return fieldDiv;
+}
+`;
+
+  // Create CSS file (empty)
+  const cssContent = `/* ${componentName.charAt(0).toUpperCase() + componentName.slice(1)} component styles */
+
+.${componentName} {
+  /* Add your custom styles here */
+}
+`;
+
+  // Create JSON file based on base component
+  let jsonContent;
+  try {
+    const baseComponentPath = path.join(__dirname, '../blocks/form/models/form-components', baseComponent.filename);
```

Comment:
pls write a basic test so that if these paths are updated in the boilerplate, then this tooling does not breaks.
right now, it assumes a lot of things.

Reply:
Added forms-scaffolder.test.js unit test